mnefun/_mnefun.py

```diff
@@ -2293,7 +2311,31 @@ def gen_covariances(p, subjects, run_indices):
                 out_rank = mne.cov.compute_whitener(cov, epochs.info,
                                                     return_rank=True,
                                                     verbose='error')[2]
-                assert want_rank == out_rank
+                if want_rank != out_rank:
+                    # Hopefully we never hit this code path, but let's keep
+                    # some debugging stuff around just in case
+                    mne.viz.plot_cov(cov, epochs.info)
+                    epochs_fnames, _ = get_epochs_evokeds_fnames(
+                        p, subj, p.analyses)
+                    epochs2 = read_epochs(epochs_fnames[1], preload=True)
+                    idx = np.searchsorted(epochs.events[:, 0],
+                                          epochs2.events[:, 0])
+                    assert len(np.unique(idx)) == len(idx)
+                    epochs = epochs[idx]
+                    assert np.array_equal(epochs.events[:, 0],
+                                          epochs2.events[:, 0])
+                    epochs2.pick_types(meg=True, eeg=True, exclude=[])
+                    import matplotlib.pyplot as plt
+                    plt.figure()
+                    for eps in (epochs, epochs2):
+                        eps = eps.get_data().transpose([1, 0, 2])
+                        eps = eps.reshape(len(eps), -1)
+                        plt.plot(np.log10(np.maximum(linalg.svdvals(eps), 1e-50)))
+                    epochs.plot()
+                    epochs2.copy().crop(p.bmin, p.bmax).plot()
```

Comment:
will this automatically `plt.show()`?  (are we guaranteed to be in `ion` mode?)

Reply:
This is just debugging code we will hopefully never need. I left it in just to make life easier when we do. But even then the default of `epochs.plot` has `show=True`